### PR TITLE
Implement intrinsic `ceiling`

### PR DESCRIPTION
**Pull request by parth121101**

### CODE REVIEW

**Review by Pranavchiku — APPROVED** [review]
Looks good to me. Thanks @parth121101 !

File: src/libasr/pass/intrinsic_function_registry.h [hunk 1]
```diff
@@ -2139,6 +2141,81 @@ namespace Floor {
 
 } // namespace Floor
 
+namespace Ceiling {
+
+    static ASR::expr_t *eval_Ceiling(Allocator &al, const Location &loc,
+            ASR::ttype_t* t1, Vec<ASR::expr_t*> &args) {
+        float val = ASR::down_cast<ASR::RealConstant_t>(args[0])->m_r;
```
Comment (Shaikh-Ubaid) [inline src/libasr/pass/intrinsic_function_registry.h:2148]: I think we should use `double` here for better accuracy.

File: src/libasr/pass/intrinsic_function_registry.h [hunk 2]
```diff
@@ -2139,6 +2141,81 @@ namespace Floor {
...
+        int result;
+        if (val <= 0.0) {
+            result = int(val);
+        } else {
+            result = int(val)+1;
+        }
```
Comment (Shaikh-Ubaid) [inline src/libasr/pass/intrinsic_function_registry.h:2154]: I think we need to check if the value is already an integer. For example, the following fails:

```console
% cat examples/expr2.f90 
program expr2
implicit none

integer(kind = 8) :: res_8

    res_8 = Ceiling(1.000, 8)
    print *, res_8
    if (res_8 /= 1) error stop

end program
% gfortran examples/expr2.f90
% ./a.out 
                    1
% lfortran examples/expr2.f90
2
ERROR STOP
```

File: src/libasr/pass/intrinsic_function_registry.h [hunk 3]
```diff
@@ -2139,6 +2141,81 @@ namespace Floor {
...
+            Vec<ASR::call_arg_t>& new_args, int64_t /*overload_id*/) {
+        declare_basic_variables("_lcompilers_Ceiling_" + type_to_str_python(arg_types[0]));
+        fill_func_arg("x", arg_types[0]);
+        auto result = declare(fn_name, return_type, ReturnVar);
+        /*
+        * r = Ceiling(x)
+        * if(x > 0.00){
+        *   r = int(x) + 1
+        * } else {
+        *   r = int(x)
```
Comment (Shaikh-Ubaid) [inline src/libasr/pass/intrinsic_function_registry.h:2202]: Similar to the previous comment, I think we need to check for integral values here as well.

File: src/libasr/pass/intrinsic_function_registry.h [hunk 4]
```diff
@@ -2139,6 +2141,81 @@ namespace Floor {
...
+namespace Ceiling {
+
+    static ASR::expr_t *eval_Ceiling(Allocator &al, const Location &loc,
+            ASR::ttype_t* t1, Vec<ASR::expr_t*> &args) {
+        float val = ASR::down_cast<ASR::RealConstant_t>(args[0])->m_r;
+        int result;
+        if (val <= 0.0) {
+            result = int(val);
+        } else {
+            result = int(val)+1;
```
Comment (Shaikh-Ubaid) [inline src/libasr/pass/intrinsic_function_registry.h:2153]: I think we need to use `int64_t` for result variable and casting here to allow for kind `8` integers.

**Review by Shaikh-Ubaid — COMMENTED** [review]
I shared some comments. Rest looks good to me. Thanks for this!

**Comment by certik** [comment]
@parth121101 can you please create issues for the above improvements that @Shaikh-Ubaid suggested? Then let's fix it.

**Pull request merged**